### PR TITLE
#1191 Integrated GraalVM PoC into ide repository

### PR DESCRIPTION
**Pull request by moritzLanger**
#1191  Integrate GraalVM PoC into ide repository

### CODE REVIEW

File: ide/src/main/java/com/devonfw/tools/ide/dev/functions/Functions.java [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+                doOpen("https://github.com/devonfw/ide/blob/master/documentation/LICENSE.asciidoc");
+            }*/
+            if (isBatch()) {
+                System.err.println("You need to accept these terms of use and all license agreements. Please rerun in interactive (non-batch) mode.");
+            }
+            boolean askToContinue = askToContinue("Do you accept these terms of use and all license agreements?", true);
+            if (askToContinue) {
+                try {
+                    licenseAgreement.createNewFile();
+                    String licenseText = "On " + LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd 'at' HH:mm:ss"))
```
[the remark this answers is not in the document]
Comment (moritzLanger) [reply ide/src/main/java/com/devonfw/tools/ide/dev/functions/Functions.java:48]: @sonatype-lift ignore

File: ide/src/main/resources/system/mac/keyboard/install-mac-keyboard-support.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+cd "$(dirname "${BASH_SOURCE:-$0}")" || exit 255
+
+if [ ! -d ~/Library/KeyBindings ]
+then
+  echo "This does not look like MacOS!"
+  exit -1
```
[the remark this answers is not in the document]
Comment (hohwille) [reply ide/src/main/resources/system/mac/keyboard/install-mac-keyboard-support.sh:8]: we should strip out these resources (`ide/src/main/resources/system`) as those just have been duplicated from ide-settings in the PoC.

File: ide/src/main/java/com/devonfw/tools/ide/dev/environment/Environment.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        propertiesPaths.add(System.getProperty("user.home")+"/devon.properties");
+        propertiesPaths.add(DEVON_IDE_HOME+"/scripts/devon.properties");
+        propertiesPaths.add(DEVON_IDE_HOME+"/settings/devon.properties");
+        propertiesPaths.add(DEVON_IDE_HOME+"/conf/devon.properties");
+        propertiesPaths.add(DEVON_IDE_HOME+"/settings/projects/*.properties");
+        for (String propertiesPath : propertiesPaths) {
+            File file = new File(propertiesPath);
+            if (file.exists()) {
+                try{
+                    this.properties.load(new FileInputStream(propertiesPath));
```
[the remark this answers is not in the document]
Comment (moritzLanger) [reply ide/src/main/java/com/devonfw/tools/ide/dev/environment/Environment.java]: fixed

File: ide/src/main/java/com/devonfw/tools/ide/dev/functions/Functions.java [hunk 4]
```diff
@@ -0,0 +1,627 @@
...
+            downloadFilename = (downloadFilename != null && !downloadFilename.isEmpty()) ? downloadFilename : downloadUrl.getPath().substring(downloadUrl.getPath().lastIndexOf('/') + 1);
+            System.out.println("Trying to download " + downloadFilename + " from " + url);
+            String target = targetFolder + downloadFilename;
+            File targetFile = new File(target);
+            if (targetFile.exists()) {
+                System.out.println("Artifact already exists at " + target + "\nTo force update please delete the file and run again.");
+                return;
+            }
+            InputStream in = new BufferedInputStream(downloadUrl.openStream());
+            FileOutputStream out = new FileOutputStream(tmpFile);
```
[the remark this answers is not in the document]
Comment (moritzLanger) [reply ide/src/main/java/com/devonfw/tools/ide/dev/functions/Functions.java]: fixed

**Review by hohwille — APPROVED** [review]
@moritzLanger thanks for your PR. 👍 
Now ready for merge.

**Pull request merged**